packaging/tags.py

```diff
@@ -433,71 +494,59 @@ def _linux_platforms(is_32bit=_32_BIT_INTERPRETER):
     manylinux_support_iter = iter(manylinux_support)
     for name, glibc_version in manylinux_support_iter:
         if _is_manylinux_compatible(name, glibc_version):
-            platforms = [linux.replace("linux", name)]
+            yield linux.replace("linux", name)
             break
-    else:
-        platforms = []
     # Support for a later manylinux implies support for an earlier version.
-    platforms += [linux.replace("linux", name) for name, _ in manylinux_support_iter]
-    platforms.append(linux)
-    return platforms
+    for name, _ in manylinux_support_iter:
+        yield linux.replace("linux", name)
+    yield linux
 
 
 def _generic_platforms():
-    # type: () -> List[str]
-    platform = _normalize_string(distutils.util.get_platform())
-    return [platform]
+    # type: () -> Iterator[str]
+    yield _normalize_string(distutils.util.get_platform())
+
+
+def _platform_tags():
+    # type: () -> Iterator[str]
+    """
+    Provides the platform tags for this installation.
+    """
+    if platform.system() == "Darwin":
+        return mac_platforms()
+    elif platform.system() == "Linux":
+        return _linux_platforms()
+    else:
+        return _generic_platforms()
 
 
 def _interpreter_name():
```

Comment:
Can we expose this one as well? Any implementation that can (optionally) take a user-provided implementation name will need to fall back to this in order to invoke the correct one of (`cpython_tags()`, `generic_tags()`) and `compatibie_tags()`

Reply:
I was somewhat waiting for someone to ask that. 😄 Yes, I think exposing this is harmless.